### ISSUE TITLE
`jdbi-bom` extends `jdbi-parent` resulting in dependency conflicts

### ISSUE THREAD

**Issue by David-16** [jdbi/jdbi#1656]
Consider the following simple `pom.xml`:

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">

  <modelVersion>4.0.0</modelVersion>

  <artifactId>foobar</artifactId>
  <groupId>foobar</groupId>
  <version>1</version>

  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.jdbi</groupId>
        <artifactId>jdbi3-bom</artifactId>
        <type>pom</type>
        <version>3.12.0</version>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>

  <dependencies>
    <dependency>
      <groupId>com.google.code.findbugs</groupId>
      <artifactId>jsr305</artifactId>
      <version>3.0.2</version>
    </dependency>
  </dependencies>
</project>
```

This pom does nothing but import the jdbi-bom and define a dependency on findbugs:jsr305

The output of `mvn dependency:tree` is:
```
[INFO] foobar:foobar:jar:1
[INFO] \- com.google.code.findbugs:jsr305:jar:3.0.2:provided
```

Note how the scope is now provided.  This results in runtime failures due to the dependency no longer being transitive.

To remedy this, `jdbi-bom` should not use `jdbi-parent` as its parent because the latter's `dependencyManagement` tag which includes a ton of 3rd party libraries specific to jdbi is leaked through the bom.

**Comment by David-16** [comment]
While the example can be fixed by adding `<scope>compile</scope>` to the `jsr305` dependency, the point is that this is not always possible/apparent when the dependency is pulled in transitively.  Plus there are dozens of such dependencies that would then need to be controlled in such a way

**Comment by stevenschlansker** [comment]
Thanks for reporting this, it'll be fixed in `3.12.1`
